crates/sui-storage/src/event_store.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! EventStore API supports high velocity event ingestion plus flexible event querying.
+//! Multiple use cases supported:
+//! - Explorer reads of different events
+//! - Filtering of events per Move package, type, or other fields
+//! - Persistent/reliable streaming, which needs to recover filtered events from a marker
+//!   or point in time
+//!   
+//! Events are also archived into checkpoints so this API should support that as well.
+//!
+
+use move_core_types::language_storage::ModuleId;
+use sui_types::base_types::TransactionDigest;
+use sui_types::event::EventEnvelope;
+
+use flexstr::SharedStr;
+use serde_json::Value;
+
+/// One event pulled out from the EventStore
+#[allow(unused)]
+struct StoredEvent {
+    /// UTC timestamp in milliseconds
+    timestamp: u64,
+    checkpoint_num: u64,
```

Comment:
is this the epoch number? if so might be better to call it `epoch_num`

Reply:
No, this is the checkpoint number.

Reply:
I thought checkpoints and epochs were in one-to-one correspondence though? is that not the case?

Reply:
I would think we would want epochs to be more frequent, and i've heard ~10 minutes thrown around as a likely number.

The point is not that the committee would change that frequently, but that there are many datastructures must be persisted for the duration of the epoch and grow monotonically - for instance `AuthorityStore::certificates`, which as I understand it must always contain all certificates that have been executed by the validator during the epoch. Hence, a longer epoch means higher storage requirements.

@gdanezis Can you comment?

Reply:
thanks - the 10 minute number i heard must have been for checkpoints, not epochs.

Reply:
@lxfind I wouldn't necessarily say that most of the storage is spent on signatures.  With batched transactions, signatures will take up a pretty insignificant amount of space compared to everything else.   There are also efforts to use smaller signatures.